app/controllers/api/v1/likes_controller.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class LikesController < Api::V1::BaseController
+      before_action only: %i[show] do
+        require_access_token %w[read]
+      end
+
+      before_action only: %i[create destroy] do
+        require_access_token %w[write]
+      end
+
+      rescue_from ActiveRecord::RecordNotFound do
+        render json: I18n.t("api.endpoint_errors.posts.post_not_found"), status: :not_found
+      end
+
+      rescue_from ActiveRecord::RecordInvalid do
+        render json: I18n.t("api.endpoint_errors.likes.user_not_allowed_to_like"), status: :not_found
+      end
+
+      def show
+        likes = like_service.find_for_post(params[:post_id])
+        render json: likes.map {|x| like_json(x) }
+      end
+
+      def create
+        begin
+          like_service.create(params[:post_id])
+        rescue ActiveRecord::RecordInvalid => e
+          if(e.message == "Validation failed: Target has already been taken")
```

Comment:
- Layout/SpaceAroundKeyword: Space after keyword `if` is missing.
- Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression.